Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -195,214 +255,169 @@ float4 SharpBilinearSample(float3 uvw, float gamma)
+    x_range = min(x_range, max_iterations);
+    y_range = min(y_range, max_iterations);
+
+    // Accumulate top and bottom edge pixels.
+    for (int ix = 0; ix < max_iterations; ++ix) {
+        if (ix < x_range) {
+            float x = f_beg.x + 1.0 + float(ix);
+            avg_color += area_n * QuickSampleByPixel(float2(x, f_beg.y) + offset, uvw.z);
+            avg_color += area_s * QuickSampleByPixel(float2(x, f_end.y) + offset, uvw.z);
+        }
+    }
+
+    // Accumulate left and right edge pixels and all the pixels in between.
+    for (int iy = 0; iy < max_iterations; ++iy) {
+        if (iy < y_range) {
+            float y = f_beg.y + 1.0 + float(iy);
+
+            avg_color += area_w * QuickSampleByPixel(float2(f_beg.x, y) + offset, uvw.z);
+            avg_color += area_e * QuickSampleByPixel(float2(f_end.x, y) + offset, uvw.z);
+
+            for (int ix = 0; ix < max_iterations; ++ix) {
+                if (ix < x_range) {
+                    float x = f_beg.x + 1.0 + float(ix);
+                    avg_color += QuickSampleByPixel(float2(x, y) + offset, uvw.z);
+                }
+            }
+        }
+    }
+
+    // Compute the area of the pixel box that was sampled.
+    float area_corners = area_nw + area_ne + area_sw + area_se;
+    float area_edges = float(x_range) * (area_n + area_s) + float(y_range) * (area_w + area_e);
+    float area_center = float(x_range) * float(y_range);
+
+    // Return the normalized average color.
+    return avg_color / (area_corners + area_edges + area_center);
 }
 
 /***** Main Functions *****/
 
 // Returns an accurate (gamma corrected) sample of a gamma space space texture.
 // Outputs in linear space for simplicity.
-float4 LinearGammaCorrectedSample(float gamma)
+float4 GetResampledColor()
 {
-	float3 uvw = v_tex0;
-	float4 color = float4(0, 0, 0, 1);
-
-	if (resampling_method <= 1) // Bilinear
-	{
-		color = BilinearSample(uvw, gamma);
-	}
-	else if (resampling_method == 2) // Bicubic: B-Spline
-	{
-		color = BicubicSample(uvw, gamma, CUBIC_COEFF_GEN(1.0, 0.0));
-	}
-	else if (resampling_method == 3) // Bicubic: Mitchell-Netravali
-	{
-		color = BicubicSample(uvw, gamma, CUBIC_COEFF_GEN(1.0 / 3.0, 1.0 / 3.0));
-	}
-	else if (resampling_method == 4) // Bicubic: Catmull-Rom
-	{
-		color = BicubicSample(uvw, gamma, CUBIC_COEFF_GEN(0.0, 0.5));
-	}
-	else if (resampling_method == 5) // Sharp Bilinear
-	{
-		color = SharpBilinearSample(uvw, gamma);
-	}
-	else if (resampling_method == 6) // Area Sampling
-	{
-		color = AreaSampling(uvw, gamma);
-	}
-	else if (resampling_method == 7) // Nearest Neighbor
-	{
-		color = QuickSample(uvw, gamma);
-	}
-	else if (resampling_method == 8) // Bicubic: Hermite
-	{
-		color = BicubicSample(uvw, gamma, CUBIC_COEFF_GEN(0.0, 0.0));
-	}
-
-	return color;
+    float3 uvw = v_tex0;
+
+    float4 color = float4(0.0, 0.0, 0.0, 1.0);
+
+    // Bilinear
+    if (resampling_method <= 1)
+        color = BilinearSample(uvw);
```

Comment:
I find not using parenthesis in branches that have spacing between each if and else unconventional